tools/hdf_dump.py

```diff
@@ -145,8 +154,14 @@ def hdf_dump_from_dataset(dataset, hdf_dataset, parser_args):
     print("Labels for %s:" % data_key, labels[:3], "...", file=log.v5)
     max_label_len = max(map(len, labels))
     if data_key != default_data_input_key:
-      hdf_dataset['targets/labels'].create_dataset(data_key, (len(labels),), dtype="S%i" % (max_label_len + 1))
-      for i, label in enumerate(labels):
+      if data_key == default_data_target_key:
```

Comment:
Instead of having this `if` branch here, I think the code would be simpler if you write this:
```
hdf_data_key = hdf_data_key_map[data_key]
```
We need to introduce `hdf_data_key_map` instead of `valid_hdf_target_key`.

Reply:
Will change it.